src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx

```diff
@@ -39,34 +43,65 @@ export const ConnectionStatus: React.FC = (): JSX.Element => {
   };
 
   const connectionInfo = useNotebook((state) => state.connectionInfo);
-  if (!connectionInfo) {
-    return <></>;
+  const memoryUsageInfo = useNotebook((state) => state.memoryUsageInfo);
+
+  const totalGB = memoryUsageInfo ? memoryUsageInfo.totalKB / 1048576 : 0;
+  const usedGB = totalGB > 0 ? totalGB - memoryUsageInfo.freeKB / 1048576 : 0;
+
+  if (
+    connectionInfo &&
+    (connectionInfo.status === ConnectionStatusType.Connect || connectionInfo.status === ConnectionStatusType.ReConnect)
+  ) {
+    return (
+      <ActionButton className="commandReactBtn" onClick={() => container.allocateContainer()}>
+        <TooltipHost content={toolTipContent}>
+          <Stack className="connectionStatusContainer" horizontal>
+            <Icon iconName="ConnectVirtualMachine" className="connectIcon" />
+            <span>{connectionInfo.status}</span>
+          </Stack>
+        </TooltipHost>
+      </ActionButton>
+    );
   }
+
   if (connectionInfo && connectionInfo.status === ConnectionStatusType.Connecting && isActive === false) {
     setIsActive(true);
+    setStatusColor("status connecting is-animating");
+    setToolTipContent("Connecting to temporary run time environment.");
   } else if (connectionInfo && connectionInfo.status === ConnectionStatusType.Connected && isActive === true) {
     stopTimer();
-    setStatusColor("locationGreenDot");
-    setStatusColorAnimation("ringringGreen");
+    setStatusColor("status connected is-animating");
+    setToolTipContent("Connected to temporary runtime environment.");
   } else if (connectionInfo && connectionInfo.status === ConnectionStatusType.Failed && isActive === true) {
     stopTimer();
-    setStatusColor("locationRedDot");
-    setStatusColorAnimation("ringringRed");
+    setStatusColor("status failed is-animating");
+    setToolTipContent("Click here to Reconnect to temporary run time environment");
   }
   return (
-    <TooltipHost content={toolTipContent}>
-      <Stack className="connectionStatusContainer" horizontal>
-        <div className="ring-container">
-          <div className={statusColorAnimation}></div>
-          <Icon iconName="LocationDot" className={statusColor} />
-        </div>
-        <span className={connectionInfo.status === ConnectionStatusType.Failed ? "connectionStatusFailed" : ""}>
-          {connectionInfo.status}
-        </span>
-        {connectionInfo.status === ConnectionStatusType.Connecting && isActive && (
-          <ProgressIndicator description={minute + ":" + second} />
-        )}
-      </Stack>
-    </TooltipHost>
+    <ActionButton
+      className={connectionInfo.status === ConnectionStatusType.Failed ? "commandReactBtn" : "connectedReactBtn"}
+      onClick={(e: React.MouseEvent<HTMLSpanElement>) =>
+        connectionInfo.status === ConnectionStatusType.Failed ? container.allocateContainer() : e.preventDefault()
```

Comment:
>Failed [](http://example.com/codeflow?start=55&length=6)

cant it be other states like default unconnected state?

Reply:
May be unconnected don't give a context to user when allocation fails.